src/zenml/integrations/mlflow/model_deployers/mlflow_model_deployer.py

```diff
@@ -0,0 +1,311 @@
+        a new one.
+
+        This method interacts with external serving platforms that manage services. 
+        In the case of MLflow, we need to look into the local filesystem at a location
+        where the configuration for the service is stored. Retrieving that, we can
+        check if there's any service that was created for the existing model and 
+        can return that or create one if there isn't any. 
+        
+        Concrete model deployer subclasses must implement the following
+        functionality in this method:
+        - Detect if there is an existing model server instance running serving
+        one or more previous versions of the same model
+        - Deploy the model to the serving platform or update the existing model
+        server instance to include the new model version
+        - Return a Service object that is a representation of the external model
+        server instance. The Service must implement basic operational state
+        tracking and lifecycle management operations for the model server (e.g.
+        start, stop, etc.)
+
+        Args:
+            pipeline_name: Name of the pipeline that the model to be deployed
+                is part of.
+            run_id: ID of the pipeline run which the model to be deployed
+                is part of.
+            step_name: The name of the pipeline model deployment step that
+                deploys the model.
+            model_name: Name of the model to be deployed.
+            model_uri: URI of the model to be deployed.
+            model_type: Type/format of the model to be deployed.
+            config: MLflow deployer config parameters passed from the step for 
+            the model deployer.
+
+        Returns:
+            The deployment Service object.
+        """
+
+        services_path = os.path.join(
+            get_global_config_directory(),
+            LOCAL_STORES_DIRECTORY_NAME,
+            str(self.uuid),
+        )
+        
+        for root, dirs, files in os.walk(services_path):
+            for file in files:
+                if(file == SERVICE_DAEMON_CONFIG_FILE_NAME):
+                    service_config_path = os.path.join(root, file)                    
+                    logger.info(
+                        "Loading service daemon configuration from %s", service_config_path
+                    )
+                    old_config = None
+                    with open(service_config_path, "r") as f:
+                        old_config = f.read()
+                    old_service = ServiceRegistry().load_service_from_json(old_config)
+                    if not isinstance(old_service, MLFlowDeploymentService):
+                        raise TypeError(
+                            f"Expected service type MLFlowDeploymentService but got "
+                            f"{type(old_service)} instead"
+                        )
+                    if(self._old_service_exists(old_service, config)):
```

Comment:
Now I'm wondering if fetching the previous version of a service is something that needs to be extracted as part of the base deployer interface and implemented by all model deployers.

Reply:
This is just a version of calling find*